sdks/python/apache_beam/runners/interactive/recording_manager.py

```diff
@@ -258,16 +278,44 @@ def _watch(self, pcolls):
         ie.current_env().watch(
             {'anonymous_pcollection_{}'.format(id(pcoll)): pcoll})
 
-  def clear(self, pcolls):
+  def _clear(self, pipeline_instrument):
     # type: (List[beam.pvalue.PCollection]) -> None
 
-    """Clears the cache of the given PCollections."""
+    """Clears the recording of all non-source PCollections."""
 
     cache_manager = ie.current_env().get_cache_manager(self.user_pipeline)
-    for pc in pcolls:
-      cache_key = self._pipeline_instrument.cache_key(pc)
+
+    # Only clear the PCollections that aren't being populated from the
+    # BackgroundCachingJob.
+    all_cached = set(
+        str(c.to_key()) for c in pipeline_instrument.cacheables.values())
```

Comment:
It's a little unsafe to use string representation as keys, no? Should CacheKey implement a hash method instead?

Reply:
It seems that we're using the string representation everywhere, so it may represent a large change - and I see it;s a carefully written repr method, so it's not necessary for this PR.

Reply:
Gotcha, I can make a follow-up PR.